docs/deployment/azure/aca-deployment-github-actions.md

```diff
@@ -169,6 +172,68 @@ The Azure Developer CLI enables you to automatically create pipelines with the c
 
 Congratulations! You successfully deployed a .NET Aspire project using the Azure Developer CLI and Azure Pipelines.
 
+## Troubleshoot Azure DevOps pipeline deployment
+
+This section covers common issues you might encounter when deploying .NET Aspire projects using Azure DevOps pipelines.
+
+### "ERROR: no project exists; to create a new project, run azd init"
+
+**Problem**: During the provisioning step of your Azure DevOps pipeline, you encounter the error message:
+
+```output
+ERROR: no project exists; to create a new project, run azd init
+```
+
+**Cause**: This error occurs because the `azd init` command generates files (`azure.yaml` and the `.azure` folder) that are typically not committed to your repository. When the pipeline runs in a clean environment, these files don't exist, causing `azd` commands to fail.
+
+**Solution**: There are several approaches to resolve this issue:
+
+#### Option 1: Run azd init in your pipeline (Recommended)
+
+Add an `azd init` step to your Azure DevOps pipeline before the provisioning step. You can use the `--from-code` and `--no-prompt` flags to run the command non-interactively:
+
+```yaml
+- task: AzureCLI@2
+  displayName: 'Initialize Azure Developer CLI'
+  inputs:
+    azureSubscription: '$(AZURE_SERVICE_CONNECTION)'
+    scriptType: 'bash'
+    scriptLocation: 'inlineScript'
+    inlineScript: |
+      azd init --from-code --no-prompt
+      azd env new $(AZURE_ENV_NAME) --location $(AZURE_LOCATION) --subscription $(AZURE_SUBSCRIPTION_ID)
+```
+
+> [!NOTE]
+> If you encounter prompts even with `--no-prompt`, try running `azd init` and `azd env new` as separate steps, or use environment variables to provide answers to any prompts. The `--from-code` flag tells azd to use the existing code in the current directory rather than creating a new project from a template.
+
+Make sure to define the following variables in your pipeline:
+
+- `AZURE_ENV_NAME`: Your environment name (for example, `dev` or `prod`)
+- `AZURE_LOCATION`: Your Azure region (for example, `eastus2`)
+- `AZURE_SUBSCRIPTION_ID`: Your Azure subscription ID
+
+#### Option 2: Commit required files to your repository
```

Comment:
@IEvangelist To me this sounds like a security problem. I don't know much about azd, but this sounds insecure.

Reply:
No, this is common. The _azure.yaml_ is just a schema for describing apps and resources. Likewise, Bicep files and _parameters.json_ are all generally safe as long as they don't include secrets and the like.